muskathlon/models/survey_user_input.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class SurveyUserInput(models.Model):
+    _inherit = 'survey.user_input'
+
+    @api.multi
+    def write(self, vals):
+        """
+        Automatically complete Medical Survey task when user filled it
+        """
+        res = super(SurveyUserInput, self).write(vals)
+        if vals.get('state') == 'done':
+
+            # Search for medical surveys
+            for registration in self.env['event.registration'].search([
+                ('medical_survey_id', 'in', self.ids),
+                ('partner_id_id', '=', self.partner_id.id)
```

Comment:
Remove this condition, as the survey will always be linked to the same partner, plus it's illegal to do `self.partner_id` because self is a recordset (containing multiple records).

Reply:
Add another condition instead checking that the event type is muskathlon.